### DISCUSSION TITLE
Specifying parameter type

### DISCUSSION THREAD

**Discussion by gputzel** [nextflow-io/nextflow#2864] [Q&A]
Is there a way to specify the type of a workflow parameter?

If I run the following at the command line:

`nextflow run main.nf --sampleID 10`

then `params.sampleID` is interpreted as an integer. I would like it to be parsed as a string. I could convert it to a string after it has been parsed as an integer, but that has the potential to cause problems (for example `--sampleID 010`).

**Comment by bentsherman** [comment]
You can define a `nextflow_schema.json` (see the nf-core pipelines for examples) but I'm not sure if this schema feature also does type casting or not.
